src/shell.js

```diff
@@ -143,6 +126,23 @@ var ENVIRONMENT_IS_PTHREAD = Module['ENVIRONMENT_IS_PTHREAD'] || false;
 var ENVIRONMENT_IS_WASM_WORKER = Module['$ww'];
 #endif
 
+// --pre-jses are emitted after the Module integration code, so that they can
+// refer to Module (if they choose; they can also define Module)
+{{{ preJS() }}}
```

Comment:
There are some reads from Module in the code this was moved over, which means if (as the comment says) a pre-JS defines Module then this PR changes the behavior. But IIUC those are internal ones like `Module['ENVIRONMENT_IS_PTHREAD']` which users do not provide anyhow, so that should be safe, is that right?

Reply:
Indeed, `ENVIRONMENT_IS_PTHREAD` is never set by a user.   As of today it gets set in `worker.js`.. once #21701 lands it will get set here in shell.js.